test/index.ts

```diff
@@ -75,6 +78,35 @@ test('server', async t => {
   t.end()
 })
 
+test('server upload', async t => {
+  const port = 5813
+  const hostname = '127.0.0.1'
+  const remotePrefix = `http://${hostname}:${port}`
+  const storeCtrlForServer = await createStoreController()
+  const server = createServer(storeCtrlForServer, {
+    hostname,
+    port,
+  })
+  const storeCtrl = await connectStoreController({remotePrefix, concurrency: 100})
+
+  const fakeEngine = 'client-engine'
+  const fakePkgId = 'test.example.com/fake-pkg/1.0.0'
+
+  await storeCtrl.upload(path.join(__dirname, 'side-effect-fake-dir'), {
+    engine: fakeEngine,
+    pkgId: fakePkgId,
+  })
+
+  const cachePath = path.join('.store', fakePkgId, 'side_effects', fakeEngine, 'package')
+  t.ok(await fs.exists(cachePath), 'cache not uploaded')
+  t.ok(await fs.exists(path.join(cachePath, 'side-effect.js')), 'cache not uploaded: missing side-effect.js')
+  t.ok(await fs.exists(path.join(cachePath, 'side-effect.txt')), 'cache not uploaded: missing side-effect.txt')
```

Comment:
the third parameter should describe what is tested. It is not a message that is printed on fail.

So it should be something like `side-effect.txt is uploaded`

Also, this test can be shorter

```js
t.deepEqual(await fs.readdir(cachePath), ['side-effect.js', 'side-effect.txt'], 'all files are uploaded')
```